src/native/clr/host/bridge-processing.cc

```diff
@@ -0,0 +1,307 @@
+	
+		// TODO this doesn't seem to be handled in the Mono code but if we don't handle this case, then the component _might not be_ strongly connected on the Java side.
+		// What is the case when this fails though? All bridge objects should have the `monodroidAddReference` method, so this is unexpected.
+		abort_unless (reference_added, [this, &prev, &next] {
+			jclass prev_java_class = env->GetObjectClass (prev->handle);
+			const char *prev_class_name = Host::get_java_class_name_for_TypeManager (prev_java_class);
+
+			jclass next_java_class = env->GetObjectClass (next->handle);
+			const char *next_class_name = Host::get_java_class_name_for_TypeManager (next_java_class);
+
+			return detail::_format_message (
+				"Failed to add reference between objects in a strongly connected component: %s -> %s.",
+				prev_class_name,
+				next_class_name);
+		});
+	
+		prev->refs_added = 1;
+		prev = next;
+	}
+}
+
+void BridgeProcessing::add_cross_reference (size_t source_index, size_t dest_index) noexcept
+{
+	CrossReferenceTarget from = select_cross_reference_target (source_index);
+	CrossReferenceTarget to = select_cross_reference_target (dest_index);
+
+	if (add_reference (from.get_handle(), to.get_handle())) {
+		from.mark_refs_added_if_needed ();
+	}
+}
+
+bool BridgeProcessing::add_reference (jobject from, jobject to) noexcept
+{
+	jclass java_class = env->GetObjectClass (from);
+	jmethodID add_method_id = env->GetMethodID (java_class, "monodroidAddReference", "(Ljava/lang/Object;)V");
+	env->DeleteLocalRef (java_class);
+
+	if (add_method_id != nullptr) {
+		env->CallVoidMethod (from, add_method_id, to);
+		return true;
+	} else {
+		env->ExceptionClear ();
+		return false;
+	}
+}
+
+void BridgeProcessing::clear_references_if_needed (JniObjectReferenceControlBlock *control_block) noexcept
+{
+	abort_if_invalid_pointer_argument (control_block, "control_block");
+	abort_unless (control_block->handle != nullptr, "Control block handle must not be null");
+	abort_unless (control_block->handle_type == JNIGlobalRefType, "Control block handle type must be global reference");
+
+	if (control_block->refs_added == 0) {
+		return;
+	}
+
+	// Clear references from the object
+	jclass java_class = env->GetObjectClass (control_block->handle);
+	jmethodID clear_method_id = env->GetMethodID (java_class, "monodroidClearReferences", "()V");
+	env->DeleteLocalRef (java_class); // Clean up the local reference to the class
+
+	if (clear_method_id != nullptr) [[likely]] {
+		env->CallVoidMethod (control_block->handle, clear_method_id);
+		control_block->refs_added = 0;
+	} else {
+		log_error (LOG_DEFAULT, "Failed to find monodroidClearReferences method");
+		env->ExceptionClear ();
+#if DEBUG
+		if (Logger::gc_spew_enabled ()) {
+			char *class_name = Host::get_java_class_name_for_TypeManager (java_class);
+			log_error (LOG_GC, "Missing monodroidClearReferences method for object of class {}", class_name);
```

Comment:
```suggestion
			log_error (LOG_GC, "Missing monodroidClearReferences method for object of class {}", optional_string (class_name));
```

`std::format` will crash on `nullptr` `char*` pointers. We can either use `std::string_view` or the `optional_string` helper method to avoid the crash.